core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -726,14 +894,23 @@ private[spark] class ExecutorAllocationManager(
       })
     }
 
-    registerGauge("numberExecutorsToAdd", numExecutorsToAdd, 0)
+    // The metrics are going to return the numbers for the default ResourceProfile.
+    // It would be nice to do include each profile somehow in the future.
```

Comment:
yes we can definitely aggregate, I went back and forth on whether to do all or just default. I'll update

Reply:
oh wait so the reason I didn't was aggregates don't make sense in some of these cases. Number of executors to add for instance.  really I think that is the only one.  The rest we can aggregate.
I could rename that one to be numberExecutorsToAddForDefaultProfile. thoughts?

Reply:
Did we ever explicitly document what numberExecutorsToAdd means ? I did not see it in our doc, but I could have missed it - if we did document it, adhering to what that means would be ideal.
Assuming not, intutively I would expect it to be total outstanding requests. Without resource profiles, default == total, so no behavior change; but with resource profile, I would have expected numberExecutorsToAdd to mean total with a resourceProfile.<rpId>.numberExecutorsToAdd (or some such naming) to give individual values per profile. Thoughts ?

Reply:
No the definition is not documented. Not sure why it never was.
I personally think this metric is odd because you have to understand how the algorithm works for this number to make sense.
Its not total outstanding requests either, its the number to add on the next iteration (not the one you are asking for now) if you still need executors.  And just because its the number to add the next iteration doesn't mean you are going to really ask for that many.
The numberTargetExecutors is what you are currently asking for. 

I agree with you that doing totals here make sense and then in the future having per profile ones that add up to the total. In this case I think that number would be useless though.

Really I'm almost more in favor of removing it as I don't really see it being very useful as a metric. thoughts on that?

I'll update the PR to do a sum for all of them and then can change/remove based on your thoughts

Reply:
I saw references to it online, which is why i was curious about existing semantics.
If it is tied to impl detail, then I agree that removing it might be a good idea.